betterproto/templates/template.py

```diff
@@ -63,11 +63,28 @@ class {{ service.py_name }}Stub(betterproto.ServiceStub):
 
     {% endif %}
     {% for method in service.methods %}
-    async def {{ method.py_name }}(self{% if method.input_message and method.input_message.properties %}, *, {% for field in method.input_message.properties %}{{ field.name }}: {% if field.zero == "None" and not field.type.startswith("Optional[") %}Optional[{{ field.type }}]{% else %}{{ field.type }}{% endif %} = {{ field.zero }}{% if not loop.last %}, {% endif %}{% endfor %}{% endif %}) -> {% if method.server_streaming %}AsyncGenerator[{{ method.output }}, None]{% else %}{{ method.output }}{% endif %}:
+    async def {{ method.py_name }}(self
+        {%- if not method.client_streaming -%}
```

Comment:
How does the templating engine handle all this extra whitespace?

Reply:
> I suspect some tests will be required for this.

Yeah, I didn't see a precedent for testing the generated services, but I can look at adding these.  I figured I'd get this out here for discussion before investing that time, in case this wasn't the direction you wanted to go.

> Does it make sense to add these without solving the problem of how to expose it to the end user?

Well, I don't have a better idea than letting the caller provider an iterator of the request objects.  It doesn't match the other methods, but I'm not sure how one would match those `unary_*` methods.  I suppose another direction could be a method that feeds into an iterator under the hood (wrapped within the method), but I don't know how you then signal the end of the method call, etc.  The request_iterator feels clean and simple and certainly works for my own use cases.

> How does the templating engine handle all this extra whitespace?

The extra whitespace is mitigated by the `{%-` stripping tag as well as the params to the jinja environment.  Finally, running everything through `black` in the end also removes any lingering artifacts, so the output is fine / matches current [albeit with the new `request_iterator` when appropriate]..

Reply:
I guess using the request_iterator argument as you describe is probably the most straightforward approach. Could also do it with a two way generator or a context provider, but neither of these is as simple to use.

I also have a PR open where I add a service test: https://github.com/danielgtaylor/python-betterproto/pull/32/commits/c762c9c549f693eced8165cb8f9e54a4f440ea55 And I think an example usage in the readme would also be necessary.

That said, I'm not a designated maintainer so can't promise anything about what will be acceptable to merge.

Reply:
@nat-n , thanks for that link to the test; that certainly looks less ceremonious than the testing setup I was seeing in grpclib itself.  Thanks!